src/qibo/tensorflow/gates.py

```diff
@@ -183,6 +188,7 @@ def __init__(self, *q: int, result: List[int] = 0):
         TensorflowGate.__init__(self)
         self.order = None
         self.ids = None
+        self.dm_result = None
```

Comment:
Please rename this attribute with the full name, so it is easier to understand.

Reply:
I renamed this to `density_matrix_result`. In the last commit I also added some checks that probabilities are between 0 and 1 in the `UnitaryChannel` gate (I noted to do this when I was running the tutorial).